src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -211,6 +198,41 @@ private async Task ProcessQueueAsync()
             }
         }
 
+        private async Task ProcessSingleWorkItemAsync(QueueItem work, int requestId)
+        {
+            var logIdName = $"{work.ClientName ?? "Default"}.{requestId++:00000000}.{work.MethodName}.{DateTime.Now:yyyy-MM-dd-HH-mm-ss}";
+            var traceSource = _logger == null ? null : await _logger.CreateTraceSourceAsync(logIdName, _cancelSource.Token).ConfigureAwait(false);
```

Comment:
do we need to create a trace source for every request, or can we use one trace source per server (for the lifetime of the server)?  I don't know how expensive it is to setup the source.

Reply:
The benefit of this approach is that is breaks out all the messages to be assocaited with a single request.

Reply:
that's fair - how easy is it to get a holistic view of the requests the server has processed?  Is each request in a different svclog file?

Reply:
> that's fair - how easy is it to get a holistic view of the requests the server has processed? Is each request in a different svclog file?

Yup.  Each request a different log file. We also use the requestId to tag each request.  So we get an entirely linearly ordered sequence showing how we processed each and every message.

Reply:
NOte: i imagine we *will* change some strategies here in the future once we try to use this to solve some problem.  I imagine until we need to use it to diagnose a real world issue, we won't know for certain what works best for investigations.

Reply:
I see, I don't know if it will be an issue in practice, but we could end up then with 100s-1000s of log files in the feedback attachments (that could potentially be interleaved with other component logs.  For an example of how they show up on feedback tickets, see the LogHub zip from the diagnostics tab on this ticket - https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1262772

I don't know if having a log file per request or if just including the requestId on all logged lines is better, so approving now and we can adjust later if needed.

Reply:
edit - I posted before refreshed and didn't see your latest comment, but I agree :)

Reply:
I see.  that's a very interesting point.  Ok.  I think this might abuse the system too much.  Let's try getting a single log.  And if we dn't like that, we can explode it out to different log files.

Reply:
You should not create a new LogId for each request. I'm pretty sure the intention is to have constants for your LogId.

To track a request's activities, you should use activity tracing and log with an XmlWriterTraceListener, which I believe LogHub helps you do automatically. 
Then be sure to activate activity tracing on the `JsonRpc` instances you create as described [here](https://github.com/microsoft/vs-streamjsonrpc/blob/master/doc/tracecontext.md). Or if you're using `ServiceJsonRpcDescriptor`, that's done for you.
I believe we require you to start an activity on the client, but as long as you do that, a nested child activity then appears on the server and you can easily view the logs with expandable nodes and everything.

If you haven't seen my demo of this, let me know and I can share it with you.

Reply:
> You should not create a new LogId for each request. 

We no longer do this. There's just one logid now.